Added extension sharing guidelines

oruebel opened the pull request:
Add page with the general guidelines for sharing extensions

oruebel committed:
Added extension sharing guidelines

bendichter commented on _pages/extension_sharing_guidelines.md:
```diff
@@ -0,0 +1,62 @@
+   1. Namespaces for extensions SHOULD generally follow the following naming conventions:
+      1. Use only lower-case ASCII letters (no special characters)
+      1. Use “-” to separate different parts of the name (no spaces allowed)
+      1. Use the naming schema “ndx-myname”, e.g., “ndx-cortical-surface”
```
My preference for names would be:

* the namespace name is e.g. "speech" (`speech.namespace.yaml`)
* the repo name is e.g. "ndx-speech" (`git clone ...ndx-speech.git`)
* the package name is e.g. "ndx_speech" (`from ndx_speech import Transcription`)

We know all non-core namespaces are going to be neurodata extensions, so there isn't much point in having the ndx prefix there. It only serves a purpose for the repo and package names, where this will be alongside other repos and packages.

oruebel replied:
I think we need to clarify this text a bit, because I think it is not clear when we refer to the name of the namespace.yaml file and name key of the namespace that is in the namespace.yaml file. My current thinking is:

* The 1) name key of the namespace, 2) the name of recipe repo in the NDX catalog, and 3) the name of package in python should all be ``ndx_speech``
* The name of 4) the yaml file of the namespace should be ``ndx_speech.namespace.yaml``

If I understand it correctly, you are proposing to change (4) the name of YAML namespace file to just ``speech.namespace.yaml`` and leave (1 -3) as is with ``ndx_speech``. I think that is probably fine. 

I'll update the text to indicate ``_`` instead of ``-`` as replacement for spaces to make sure we can have the same name for the python package and repos and namespace name.

oruebel replied:
@bendichter I updated the text for "Naming namespaces and repositories". Does this look better?

oruebel committed:
Updated naming guidelines for namespaces

rly committed:
Update naming rules

rly commented:
I updated the naming rules based on the discussion here: https://github.com/nwb-extensions/ndx-template/issues/21

We can change it back, but my preference is:
1. the name key of the namespace should be ndx-speech
2. the name of the source repo and extension on PyPI is ndx-speech
3. the name of the record repo in the NDX catalog should be ndx-speech-record
4. the name of the package in python should be ndx_speech (hyphens not allowed)
5. the name of the namespace yaml file should be ndx-speech.namespace.yaml

Please see my reasoning in the above link.

rly committed:
Merge branch 'master' into add/sharingguide

rly committed:
Change sharing guidelines name to sharing_guidelines

rly committed:
Fix spelling

oruebel committed:
Fix URL in for sharing guideline in main menu

oruebel committed:
Updated terms to match the sharing strategies document and some text clean up

oruebel commented:
@rly I updated the terms to match and did some minor clean up. I think this one is good to go.

rly committed:
Update sharing_guidelines.md

Require README file for release of extensions
Mention that registration within the NDX catalog requires the name key of the namespace to begin with "ndx-"
Do not say that the ndx-custom repository SHOULD include a URL (what would this URL point to?)

rly committed:
Add link to versioning, NWB:N -> NWB

rly commented:
I made a few changes:
- Require README file for release of extensions
- Mention that registration within the NDX catalog requires the name key of the namespace to begin with "ndx-"
- Do not say that the ndx-custom repository SHOULD include a URL (what would this URL point to?)
- Minor cleanup

I think this is good to go. Feel free to merge.

rly committed:
FIx typo

rly committed:
Merge branch 'master' into add/sharingguide
